tests/test_diffraction_fft.py

```diff
@@ -0,0 +1,477 @@
+    img_b = np.ones((7, 7), dtype=int)
+    diffraction_fft.add_image(img_a, img_b, -1, 0)
+    expected_img = np.ones((5, 6), dtype=int)
+    np.testing.assert_array_equal(img_a, expected_img)
+
+
+def test_saturated_clusters_works_for_no_saturated_pixels():
+    img = np.zeros((5, 6), dtype=int)
+    regions = diffraction_fft.saturated_clusters(img, 0.5)
+    assert not regions
+
+
+def test_saturated_clusters_works_for_1_saturated_pixel():
+    img = np.zeros((5, 6), dtype=int)
+    img[2, 3] = 1.0
+    regions = diffraction_fft.saturated_clusters(img, 0.5)
+    assert regions == [(slice(2, 3), slice(3, 4))]
+
+
+def test_saturated_clusters_finds_saturated_clusters():
+    img = np.array(
+        [
+            [0, 1, 0, 0, 0, 0],
+            [0, 0, 0, 0, 1, 1],
+            [1, 0, 1, 0, 0, 0],
+            [0, 1, 0, 0, 0, 1],
+            [1, 0, 1, 0, 0, 1],
+        ]
+    )
+    regions = diffraction_fft.saturated_clusters(img, 0.5)
+    expected_regions = [
+        (slice(0, 1), slice(1, 2)),
+        (slice(1, 2), slice(4, 6)),
+        (slice(2, 5), slice(0, 3)),
+        (slice(3, 5), slice(5, 6)),
+    ]
+    assert sorted(regions) == sorted(
+        expected_regions
+    ), f"{sorted(regions)} != {sorted(expected_regions)}"
+
+
+def test_fft_diffraction_is_similar_to_raytracing_for_0_exptime():
+    """Check, that raytracing diffraction and FFT diffraction are "similar"."""
+    config = create_test_config(XSIZE, YSIZE, STAMP_SIZE, exptime=0.0)
+    # To compare with the raytracing method, instead use:
+    # config = create_test_config(XSIZE, YSIZE, STAMP_SIZE, exptime=0.0, mode=Mode.RAYTRACING)
+
+    with assert_no_error_logs() as logger:
+        image = galsim.config.BuildImage(config, logger=logger)
+    # To save the image produced here, use:
+    # galsim.fits.write(image, file_name="/tmp/spikes.fits")
+    brightness = image.array
+    # Center of star:
+    [c_x, c_y] = center_of_brighness(brightness)
+    # 2 Pixel tolerance:
+    np.testing.assert_allclose(
+        np.array([c_x, c_y]), np.array([1923.0, 3161.0]), atol=2.0, rtol=0.0
+    )
+    angle, angle_stddev = folded_spike_angle(brightness, c_x, c_y, r_min=10.0)
+    expected_angle = (
+        45.0 * galsim.degrees - config["stamp"]["diffraction_psf"]["rottelpos"]
+    ).deg
+    # 1° tolerance:
+    np.testing.assert_allclose(np.rad2deg(angle), expected_angle, atol=1.0, rtol=0.0)
+    expected_max_spike_width = 5.0
+    np.testing.assert_array_less(np.rad2deg(angle_stddev), expected_max_spike_width)
+    slope, intercept, slope_stderr, intercept_stderr = brightness_log_profile(
+        brightness, c_x, c_y
+    )
+    np.testing.assert_allclose(slope, -2.0, atol=0.6, rtol=0.0)
+    np.testing.assert_allclose(intercept, 12.5, atol=3.0, rtol=0.0)
+    # Assert some rough evidence, that brightness decays as
+    # brightness ~ exp(intercept) * r**slope:
+    np.testing.assert_array_less(slope_stderr, 0.2)
+    np.testing.assert_array_less(intercept_stderr, 0.8)
```

Comment:
I don't see how these tests match the comment that it's checking "brightness ~ exp(intercept) * r**slope"